packages/app/src/DashboardFiltersEditModal.tsx

```diff
@@ -0,0 +1,284 @@
+                    ))}
+                  </Group>
+                </Radio.Group>
+              )}
+            />
+          </Input.Wrapper>
+        )}
+
+        <Input.Wrapper
+          label="Filter Expression"
+          description="SQL column or expression to filter on"
+          required
+        >
+          <SQLInlineEditorControlled
+            tableConnections={tableConnection}
+            control={control}
+            name="expression"
+            placeholder="SQL column or expression"
+            language="sql"
+            enableHotkey
+            rules={{ required: true }}
+            parentRef={parentRef}
+          />
+        </Input.Wrapper>
+
+        <Group justify="space-between" mt="md">
+          <Button
+            variant="outline"
+            color="red"
+            onClick={() => onRemoveFilter(filter.id)}
+          >
+            Delete
+          </Button>
+          <Button type="submit" className="align-self-end">
+            Save
+          </Button>
+        </Group>
+      </Stack>
+    </form>
+  );
+};
+
+interface EmptyStateProps {
+  onCreateFilter: () => void;
+}
+
+const EmptyState = ({ onCreateFilter }: EmptyStateProps) => {
+  return (
+    <Stack align="center" justify="center" py="xl">
+      <Text size="md" maw={300} ta="center">
+        Dashboard filters allow users of this dashboard to quickly filter on
+        important columns. Filters are saved in the dashboard.
+      </Text>
+      <Button variant="outline" onClick={onCreateFilter}>
+        Create Filter
+      </Button>
+    </Stack>
+  );
+};
+interface DashboardFiltersEditModalProps {
+  opened: boolean;
+  onClose: () => void;
+  filters: DashboardFilter[];
+  onSaveFilter: (filter: DashboardFilter) => void;
+  onRemoveFilter: (id: string) => void;
+  parentRef?: HTMLElement | null;
+}
+
+const NEW_FILTER_ID = 'new';
+
+const DashboardFiltersEditModal = ({
+  opened,
+  onClose,
+  filters,
+  onSaveFilter,
+  onRemoveFilter,
+}: DashboardFiltersEditModalProps) => {
+  const [selectedFilter, setSelectedFilter] = useState<
+    DashboardFilter | undefined
+  >(filters[0]);
+
+  useEffect(() => {
+    if (opened) {
+      setSelectedFilter(filters[0]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [opened]);
+
+  const handleRemoveFilter = (id: string) => {
+    if (id === selectedFilter?.id) {
+      setSelectedFilter(filters.find(f => f.id !== id));
+    }
+    onRemoveFilter(id);
+  };
+
+  const handleAddNewFilter = () => {
+    setSelectedFilter({
+      id: NEW_FILTER_ID,
+      type: 'QUERY_EXPRESSION',
+      name: 'New Filter',
+      expression: '',
+      source: '',
+    });
+  };
+
+  const handleSaveFilter = (filter: DashboardFilter) => {
+    if (filter.id === NEW_FILTER_ID) {
+      const filterWithRealId = { ...filter, id: crypto.randomUUID() };
+      onSaveFilter(filterWithRealId);
+      setSelectedFilter(filterWithRealId);
```

Comment:
After hitting save, I was sitting there thinking it didn't work, I didn't realize it has created the filter. The form should either clear and we show a toast or something, or we should highlight the selected filter in the sidebar, not sure, I didn't realize it was working